src/jcstress/java/reactor/pool/SimpleDequePoolStressTest.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2018-Present Pivotal Software Inc, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package reactor.pool;
+
+import java.time.Duration;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.openjdk.jcstress.annotations.Actor;
+import org.openjdk.jcstress.annotations.Arbiter;
+import org.openjdk.jcstress.annotations.JCStressTest;
+import org.openjdk.jcstress.annotations.Outcome;
+import org.openjdk.jcstress.annotations.State;
+import org.openjdk.jcstress.infra.results.LI_Result;
+
+import reactor.core.publisher.Mono;
+
+import static org.openjdk.jcstress.annotations.Expect.ACCEPTABLE;
+import static org.openjdk.jcstress.annotations.Expect.FORBIDDEN;
+
+public class SimpleDequePoolStressTest {
+
+	@JCStressTest
+	@Outcome(id = {"ACQUIRED '200', 0"}, expect = ACCEPTABLE,  desc = "acquired")
+	@Outcome(id = {"NOT ACQUIRED, 1"}, expect = ACCEPTABLE,  desc = "acquired failed fast during/before destroy")
+	@Outcome(id = {"ACQUIRED '1', 0"}, expect = FORBIDDEN,  desc = "acquired the first resource, before destroy")
+	@Outcome(id = {"ACQUIRED '1234', 0"}, expect = FORBIDDEN,  desc = "acquired the first resource, after destroy")
+	@State
+	public static class MaxPendingAcquireWhileDestroying {
+
+		final AtomicBoolean firstResourceCreated = new AtomicBoolean();
+
+		final SimpleDequePool<AtomicInteger> pool = PoolBuilder
+				.from(Mono.defer(() -> {
+					if (firstResourceCreated.getAndSet(true)) {
+						return Mono.just(new AtomicInteger(200));
+					}
+					else {
+						return Mono.just(new AtomicInteger(1));
+					}
+				}))
+				.sizeBetween(0, 1)
+				.maxPendingAcquire(0)
+				.releaseHandler(ai -> Mono.fromRunnable(ai::incrementAndGet))
+				.evictionPredicate((res, meta) -> {
+					int v = res.get();
+					return v > 0 && v != 111;
+				})
+				.destroyHandler(ai -> Mono.fromRunnable(() -> ai.addAndGet(1233)))
+				.build(conf -> new SimpleDequePool<>(conf, true));
+
+		private final PooledRef<AtomicInteger> ref = pool.acquire().block(Duration.ofMillis(100));
+
+		@Actor
+		public void invalidate() {
+			ref.invalidate().block(Duration.ofMillis(100));
+		}
+
+		@Actor
+		public void acquisition(LI_Result r) {
+			try {
+				AtomicInteger ai = pool.acquire().block().poolable();
+				r.r1 = "ACQUIRED '" + ai.get() + "'";
```

Comment:
jcstress tests are usually very sensitive to code that actors run. It is unconventional to use `String` results here. At very least, that slows down the tests, which makes less samples per time unit, and thus lowers the probability to catch the errors. I'd suggest taking `r1 = ai.get()`, and then doing `r1 = -1` on the exceptional path.

Reply:
makes sense, yeah. the forbidden/acceptable results look better but if that is at the expense of the reliability of the test that's not a great tradeoff. that's what `desc` is about after all. will look into changing to `II_Result`.